src/pages/workspace/accounting/qbd/QuickBooksDesktopSetupPage.tsx

```diff
@@ -0,0 +1,88 @@
+import type {StackScreenProps} from '@react-navigation/stack';
+import React, {useEffect, useState} from 'react';
+import {View} from 'react-native';
+import Computer from '@assets/images/laptop-with-second-screen-sync.svg';
+import Button from '@components/Button';
+import CopyTextToClipboard from '@components/CopyTextToClipboard';
+import FixedFooter from '@components/FixedFooter';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import ImageSVG from '@components/ImageSVG';
+import ScreenWrapper from '@components/ScreenWrapper';
+import Text from '@components/Text';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import * as QuickBooksDesktop from '@libs/actions/connections/QuickBooksDesktop';
+import Navigation from '@libs/Navigation/Navigation';
+import type {SettingsNavigatorParamList} from '@libs/Navigation/types';
+import LoadingPage from '@pages/LoadingPage';
+import ROUTES from '@src/ROUTES';
+import type SCREENS from '@src/SCREENS';
+
+type RequireQuickBooksDesktopModalProps = StackScreenProps<SettingsNavigatorParamList, typeof SCREENS.WORKSPACE.ACCOUNTING.QUICKBOOKS_DESKTOP_SETUP_REQUIRED_DEVICE_MODAL>;
+
+function RequireQuickBooksDesktopModal({route}: RequireQuickBooksDesktopModalProps) {
+    const {translate} = useLocalize();
+    const styles = useThemeStyles();
+    const policyID: string = route.params.policyID;
+    const [isLoading, setIsLoading] = useState(true);
+    const [codatSetupLink, setCodatSetupLink] = useState<string>('');
+
+    useEffect(() => {
+        const fetchSetupLink = () => {
+            // eslint-disable-next-line rulesdir/no-thenable-actions-in-views
```

Comment:
Could we use `await` instead of `.then` and make `getQuickbooksDesktopCodatSetupLink()` an async function?

Reply:
Understood, thanks for the discussions link.